### PR TITLE
Equals container bugfix

### PR DESCRIPTION
**Pull request by xtonik**
### SUMMARY
There was missing code for equality test between array and bitmap containers. Above that, equality between run and bitmap containers optimized. Changes affects also mappable variants of containers.

### Automated Checks

- [ ] I have run `./gradlew test` and made sure that my PR does not break any unit test.
- [ ] I have run `./gradlew checkstyleMain` or the equivalent and corrected the formatting warnings reported.

### CODE REVIEW

**Comment by lemire** [comment]
> There was missing code for equality test between array and bitmap containers.

An array container cannot be equal to a bitmap container in the Roaring format. If you think that there is a bug, please provide a test case.

File: RoaringBitmap/src/main/java/org/roaringbitmap/ArrayContainer.java [hunk 1]
```diff
@@ -375,6 +375,8 @@ public boolean equals(Object o) {
       return ArraysShim.equals(this.content, 0, cardinality, srb.content, 0, srb.cardinality);
     } else if (o instanceof RunContainer) {
       return o.equals(this);
+    } else if (o instanceof BitmapContainer) {
```
Comment (lemire) [inline RoaringBitmap/src/main/java/org/roaringbitmap/ArrayContainer.java:378]: This cannot happen. An ArrayContainer can never be equal to a BitmapContainer when the data abides by the specification.

File: RoaringBitmap/src/main/java/org/roaringbitmap/BitmapContainer.java [hunk 2]
```diff
@@ -423,6 +423,19 @@ public boolean equals(Object o) {
       return Arrays.equals(this.bitmap, srb.bitmap);
     } else if (o instanceof RunContainer) {
       return o.equals(this);
+    } else if (o instanceof ArrayContainer) {
```
Comment (lemire) [inline RoaringBitmap/src/main/java/org/roaringbitmap/BitmapContainer.java:426]: This cannot happen. An ArrayContainer can never be equal to a BitmapContainer when the data abides by the specification.

File: RoaringBitmap/src/main/java/org/roaringbitmap/buffer/MappeableArrayContainer.java [hunk 3]
```diff
@@ -449,6 +449,8 @@ public boolean equals(Object o) {
       return true;
     } else if (o instanceof MappeableRunContainer) {
       return o.equals(this);
+    } else if (o instanceof MappeableBitmapContainer) {
```
Comment (lemire) [inline RoaringBitmap/src/main/java/org/roaringbitmap/buffer/MappeableArrayContainer.java:452]: This cannot happen. An ArrayContainer can never be equal to a BitmapContainer when the data abides by the specification.

**Comment by lemire** [comment]
The issue here is that an array container can have up to `DEFAULT_MAX_SIZE` elements, and then it must become either a bitmap or run container. That's why we don't need to compare an array with a bitmap container.

The other optimization is maybe interesting. :-)

**Comment by xtonik** [comment]
As you said, this can happen, when obscure way is used to create containers only. Above that, some optimization is included also, which is not optimization at all. Closing as irrelevant PR.

**Comment by xtonik** [comment]
The conclusion of this PR contradicts with #29.

**Pull request closed without merging**